src/app/pages/Swap/form/SwapFormInput/index.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { FC, useCallback, useMemo } from 'react';
+
+import BigNumber from 'bignumber.js';
+
+import SwapInput from 'app/pages/Swap/form/SwapFormInput/SwapInput';
+import SwapInputHeader from 'app/pages/Swap/form/SwapFormInput/SwapInputHeader';
+import { InputContainer } from 'app/templates/InputContainer/InputContainer';
+import { useFormAnalytics } from 'lib/analytics';
+import { TEZ_TOKEN_SLUG } from 'lib/assets';
+import { parseChainAssetSlug } from 'lib/assets/utils';
+import { useTezosAssetBalance } from 'lib/balances';
+import { useAssetFiatCurrencyPrice, useFiatCurrency } from 'lib/fiat-currency';
+import {
+  useCategorizedTezosAssetMetadata,
+  useGetCategorizedAssetMetadata,
+  AssetMetadataBase,
+  useTezosTokensMetadataPresenceCheck
+} from 'lib/metadata';
+import { useAvailableRoute3TokensSlugs } from 'lib/route3/assets';
+import { useBooleanState } from 'lib/ui/hooks';
+
+import { SwapSelectAssetModal } from '../../modals/SwapSelectAsset';
+
+import { SwapFormInputProps } from './SwapFormInput.props';
+
+const EXCHANGE_XTZ_RESERVE = new BigNumber('0.3');
+
+/** @deprecated // Bad practice */
+const DEFAULT_ASSET_METADATA: AssetMetadataBase = {
+  decimals: 0,
+  symbol: '',
+  name: '',
+  thumbnailUri: ''
+};
+
+const SwapFormInput: FC<SwapFormInputProps> = ({
+  network,
+  publicKeyHash,
+  className,
+  error,
+  value: { assetSlug, amount },
+  label,
+  name,
+  amountInputDisabled,
+  testIDs,
+  onChange,
+  shouldUseFiat = false,
+  setShouldUseFiat = () => {}
+}) => {
+  const { trackChange } = useFormAnalytics('SwapForm');
+
+  const isTezosSlug = assetSlug === TEZ_TOKEN_SLUG;
+  const assetSlugWithFallback = assetSlug ?? TEZ_TOKEN_SLUG;
+
+  // eslint-disable-next-line no-type-assertion/no-type-assertion
```

Comment:
pipelines will pass without this eslint rule, we have only warn on type assertion, but try to use them as few as possible